### PR TITLE
Shindo to Minitest: compute

### PR DESCRIPTION
**Pull request by gildub**
Migrates compute models and requests
    
Tests marked has `skip`(pending) using Minitest::Test::UNIT_TESTS_CLEAN
- 1 x requests/compute/security_group_tests.rb
- 1 x requests/compute/server_tests.rb
    
Besides later 2 to be addressed the random tests failures have not occurred so far!
    
Note: This is not using subfolder 'openstack' anymore

### CODE REVIEW

File: fog-openstack.gemspec [hunk 1]
```diff
@@ -33,4 +33,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'minitest'
   spec.add_development_dependency 'vcr'
   spec.add_development_dependency 'webmock',  '~> 1.24.6'
+  spec.add_development_dependency 'pry-byebug'
```
Comment (petrblaho) [inline fog-openstack.gemspec]: Should this be here?

File: test/models/compute/images_tests.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require "test_helper"
+
+describe "Fog::Compute[:openstack] | images collection" do
+  describe "success" do
+    it "#all" do
+      fog = Fog::Compute[:openstack]
+      fog.images.all.must_be_kind_of Array
+      fog.images.all.wont_be_nil
```
Comment (petrblaho) [inline test/models/compute/images_tests.rb]: Can you please split these to asserts into separate `it`s?

File: test/models/compute/security_group_tests.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        :name        => 'my_group',
+        :description => 'my group'
+      )
+    end
+
+    it "#create" do
+      @security_group.name.must_equal 'my_group'
+      @security_group.description.must_equal 'my group'
+      @security_group.security_group_rules.must_equal []
+      @security_group.tenant_id.wont_be_nil
```
Comment (petrblaho) [inline test/models/compute/security_group_tests.rb]: DITTO as above.

File: test/models/compute/security_group_tests.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          :ip_range        => {"cidr" => "0.0.0.0/0"}
+        )
+        rule.destroy
+        rule.reload.must_equal nil
+      end
+    end
+
+    after do
+      @security_group.destroy if @security_group
+    end
```
Comment (petrblaho) [inline test/models/compute/security_group_tests.rb]: Can you move `after` block into beginning of its `describe` (preferable after `before` block) please?

File: test/models/compute/server_tests.rb [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+          server.destroy if server
+
+          begin
+            @compute.servers.get(server.id).wait_for { false }
+          rescue Fog::Errors::Error
+            # ignore, server went away
+          end
+        end
+        my_group.destroy if my_group
+      end
```
Comment (petrblaho) [inline test/models/compute/server_tests.rb]: This feels quite strange to have all this code inside of `it` block especially with `begin`-`ensure`-`end`. Also this `it` has more then one assert inside.

File: test/models/compute/server_tests.rb [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+        unless Fog.mocking?
+          server.destroy if server
+
+          begin
+            @compute.servers.get(server.id).wait_for { false }
+          rescue Fog::Errors::Error
+            # ignore, server went away
+          end
+        end
+      end
```
Comment (petrblaho) [inline test/models/compute/server_tests.rb]: DITTO

File: test/models/compute/server_tests.rb [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
```
Comment (petrblaho) [inline test/models/compute/server_tests.rb]: DITTO

File: test/models/compute/server_tests.rb [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+          begin
+            @compute.servers.get(server.id).wait_for { false }
+            @compute.volumes.get(volume.id).wait_for { false }
+          rescue Fog::Errors::Error
+            # ignore, server went away
+          end
+        end
+      end
+    end
```
Comment (petrblaho) [inline test/models/compute/server_tests.rb]: DITTO

File: test/requests/compute/aggregate_tests.rb [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        must_match_schema('aggregate' => @aggregate_format)
+    end
+
+    it "#get_aggregate" do
+      @compute.get_aggregate(@aggregate['id']).body.
+        must_match_schema('aggregate' => @detailed_aggregate_format)
+    end
+
+    it "#add_aggregate_host" do
+      @host_name = @compute.hosts.select { |x| x.service_name == 'compute' }.first.host_name
```
Comment (petrblaho) [inline test/requests/compute/aggregate_tests.rb]: `host_name` do not have to be instance variable.

**Comment by petrblaho** [comment]
I am little concerned about big chunks of code in `it` blocks in `test/models/compute/server_tests.rb`.

But I understand that this PR is about migrating tests so I am ok with that. We can clean it later.

Thanks for this work. :+1:

File: fog-openstack.gemspec (hunk 1 above)
Comment (gildub) [inline fog-openstack.gemspec]: Not this time, but I'm thinking of adding it for good! :)

File: test/models/compute/security_group_tests.rb (hunk 4 above)
Comment (gildub) [inline test/models/compute/security_group_tests.rb]: Done.

File: test/models/compute/security_group_tests.rb (hunk 3 above)
Comment (gildub) [inline test/models/compute/security_group_tests.rb]: Done.

File: test/models/compute/images_tests.rb (hunk 2 above)
Comment (gildub) [inline test/models/compute/images_tests.rb]: Done.

File: test/models/compute/server_tests.rb (hunk 5 above)
Comment (gildub) [inline test/models/compute/server_tests.rb]: Replaced with #after and proper #it break down.

File: test/models/compute/server_tests.rb (hunk 6 above)
Comment (gildub) [inline test/models/compute/server_tests.rb]: Done

File: test/models/compute/server_tests.rb (hunk 7 above)
Comment (gildub) [inline test/models/compute/server_tests.rb]: Done too.

File: test/requests/compute/aggregate_tests.rb (hunk 9 above)
Comment (gildub) [inline test/requests/compute/aggregate_tests.rb]: Fixed.

File: test/models/compute/server_tests.rb (hunk 8 above)
Comment (gildub) [inline test/models/compute/server_tests.rb]: Done.

**Comment by gildub** [comment]
@petrblaho , I have addressed all issues, thanks for helping.

**Pull request merged**